openassessment/xblock/test/test_openassessment.py

```diff
@@ -91,95 +98,252 @@ def test_student_view_workflow_error(self, xblock):
-        self.assertTrue(submission_response.body.find("April"))
-
-    @scenario('data/basic_scenario.xml')
-    def test_formatted_dates(self, xblock):
-
-        # Set start/due dates
-        xblock.start = dt.datetime(2014, 4, 1, 1, 1, 1)
-        xblock.due = dt.datetime(2014, 5, 1)
-
-        request = namedtuple('Request', 'params')
-        request.params = {}
-        resp = xblock.render_peer_assessment(request)
-        self.assertTrue(resp.body.find('Tuesday, April 01, 2014'))
-        self.assertTrue(resp.body.find('Thursday, May 01, 2014'))
-
-    @scenario('data/basic_scenario.xml')
-    def test_formatted_dates_for_beta_tester_with_days_early(self, xblock):
-        """Test dates for beta tester with days early"""
-
-        # Set start/due dates
-        xblock.start = dt.datetime(2014, 4, 6, 1, 1, 1)
-        xblock.due = dt.datetime(2014, 5, 1)
-        xblock.xmodule_runtime = Mock(
-            course_id='test_course',
-            anonymous_student_id='test_student',
-            days_early_for_beta=5,
-            user_is_staff=False,
-            user_is_beta_tester=True
-        )
-        self.assertEqual(xblock.xmodule_runtime.days_early_for_beta, 5)
-        request = namedtuple('Request', 'params')
-        request.params = {}
-        resp = xblock.render_peer_assessment(request)
-        self.assertTrue(resp.body.find('Tuesday, April 01, 2014'))
-        self.assertTrue(resp.body.find('Thursday, May 01, 2014'))
+        with patch('openassessment.xblock.submission_mixin.get_current_time_zone') as time_zone_fn:
+            time_zone_fn.return_value = pytz.timezone(time_zone)
 
+            xblock = self.load_scenario('data/dates_scenario.xml')
+            xblock_fragment = self.runtime.render(xblock, "student_view")
+            self.assertIn("OpenAssessmentBlock", xblock_fragment.body_html())
+
+            # Validate Submission Rendering.
+            submission_response = xblock.render_submission({})
+            self.assertIsNotNone(submission_response)
+            self.assertIn("openassessment__response", submission_response.body)
+            self.assertIn(expected_date, submission_response.body)
+
+    @ddt.data(('utc', 'April 1, 2014 01:01 UTC'),
+              ('America/Los_Angeles', 'March 31, 2014 18:01 PDT'))
+    @ddt.unpack
+    @freeze_time("2014-01-01")
+    def test_formatted_start_dates(self, time_zone, expected_start_date):
+        """Test start dates correctly formatted"""
+        with patch(self.TIME_ZONE_FN_PATH) as time_zone_fn:
+            time_zone_fn.return_value = pytz.timezone(time_zone)
+
+            # Set start dates'
+            xblock = self.load_scenario('data/basic_scenario.xml')
+            xblock.start = dt.datetime(2014, 4, 1, 1, 1, 1)
+            request = namedtuple('Request', 'params')
+            request.params = {}
+            resp = xblock.render_peer_assessment(request)
+
+            self.assertIn(expected_start_date, resp.body)
```

Comment:
These test methods can be DRYed up by creating a few helper methods.  This way, it makes it better for code maintenance - rather than having duplicate code - even if (especially if) it's test code.

So, for example, creating helper methods for (1) setting start/due dates and (2) setting up `days_early_for_beta`.